### PR TITLE
[ECO-5015] Implement public API changes of CHADR-062

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
Implements the changes to the public API which we decided in [CHADR-062](https://ably.atlassian.net/wiki/spaces/CHA/pages/3410526209/CHADR-062+Renaming+Lifecycle+Types+for+Clarity). Highlights:

- removal of the intermediate `Status` objects for `Room` and `Connection`, putting access to the status directly on the room / connection itself
- renaming the `*Lifecycle` enums to `*Status`

## Release Notes

- **New Features**
  - Enhanced room status management with asynchronous updates and structured notifications for status changes.
  - Introduced continuous rotation effects for reactions in chat.

- **Bug Fixes**
  - Improved error handling and clarity in status transitions within the room lifecycle.

- **Tests**
  - Updated tests to reflect changes in room status management and added new tests for verifying initial states and transitions.
  - Enhanced clarity and organization of test cases related to room status.

- **Documentation**
  - Terminology updated throughout the codebase for consistency, replacing "lifecycle" with "status."

### CODE REVIEW

File: Sources/AblyChat/Room.swift [hunk 1]
```diff
@@ -97,6 +106,24 @@ internal actor DefaultRoom: Room {
...
+        transition(to: .detached)
+    }
+
+    // MARK: - Room status
+
+    internal func onStatusChange(bufferingPolicy: BufferingPolicy) -> Subscription<RoomStatusChange> {
+        let subscription: Subscription<RoomStatusChange> = .init(bufferingPolicy: bufferingPolicy)
+        statusSubscriptions.append(subscription)
+        return subscription
+    }
```
[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply Sources/AblyChat/Room.swift:118]: this code has just been moved from elsewhere

File: Tests/AblyChatTests/DefaultRoomTests.swift [hunk 2]
```diff
@@ -126,4 +130,37 @@ struct DefaultRoomTests {
...
+        await room.transition(to: newStatus)
+
+        // Then: It emits a status change to all subscribers added via onChange(bufferingPolicy:), and updates its `status` property to the new state
+        for statusChange in try await [#require(statusChange1), #require(statusChange2)] {
+            #expect(statusChange.previous == originalStatus)
+            #expect(statusChange.current == newStatus)
+        }
+
+        #expect(await room.status == .attached)
+    }
```
[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply Tests/AblyChatTests/DefaultRoomTests.swift:165]: this code has just been moved from elsewher

**Comment by lawrence-forooghian** [comment]
Will rebase and fix conflicts.

**Comment by lawrence-forooghian** [comment]
OK, fixed.

**Review by maratal — APPROVED** [review]
LGTM

**Pull request merged**